rsp/xxx_rest_api_design.md

```diff
@@ -0,0 +1,261 @@
+- Feature name: Lycan REST API v1
+- Start Date: 2016-03-09
+- RSP number: (leave this empty)
+
+Lycan REST API v1
+=================
+
+API base:
+
+/api/v1
+
+All methods are authenticated using a Access-Token HTTP header.
+
+A request without the proper header, or with invalid header will return an error code
+401: Unauthorized
+
+XXX: How to get the access token?
+
+## GET /maps
+
+Use this method to get available maps
+
+XXX: What is the data associated with a map?
+
+### Returns
+
+```javascript
+[
+{
+    "id": 0
+},
+{
+    "id": 1
+}
+]
+```
+
+## GET /maps/:id/instances
+
+Get running instances for a particular map
+
+### Returns
+
+```javascript
+[
+{
+    "id": 2,
+    "map": 0,
+    "uptime": 1000
+},
+{
+    "id": 3,
+    "map": 0,
+    "uptime": 30
+}
+]
+```
+
+## GET /instances/:id/entities
+
+Get entities in a particular instance
+
+Parameters:
+* `type`: Restricts search to the specified type ("monster" or "player")
+
+### Returns
+
+```javascript
+[
+{
+    "id": 4,
+    "type": {
+        "player": {
+            "id": 1,
+            "name": "Vaelden",
+            "gold": 42,
+            "experience": 43,
+            "guild": "God"
+        }
+    },
+    "skin": 1,
+    "current_pv": 100,
+    "position": {
+        "x": 0.1,
+        "y": 42.42,
+        "map": 0,
+        "instance": 2
+    },
+    "stats": {
+        "level": 1,
+        "strength": 2,
+        "dexterity": 3,
+        "constitution": 4,
+        "intelligence": 5,
+        "precision": 6,
+        "wisdom": 7
+    }
+},
+{
+    "id": 5,
+    "type": {
+        "monster": {
+            "id": 3,
+            "name": "zombie",
+            "behaviour_tree": "zombie.bt"
+        }
+    },
+    "skin": 0,
+    "current_pv": 100,
+    "position": {
+        "x": 0.0,
+        "y": 10.0,
+        "map": 0,
+        "instance": 3
+    },
+    "stats": {
+        "level": 7,
+        "strength": 6,
+        "dexterity": 5,
+        "constitution": 4,
+        "intelligence": 3,
+        "precision": 2,
+        "wisdom": 1
+    }
+}
+]
+```
+
+## POST /instances/:id/spawn
+
+Spawn a specific mob in the instance
+
+### Body
+
+```javascript
+{
+    "monster_class": 3,
```

Comment:
If we think long term, `monster_class` will probably be unique across many API including in the editor where it will be first generated. Maybe choosing a uuid format would be better?